diff --git a/third_party/sqlite3/ext/sqlite-vec/sqlite-vec.c b/third_party/sqlite3/ext/sqlite-vec/sqlite-vec.c
index 96612e2..7ae2afa 100644
--- a/third_party/sqlite3/ext/sqlite-vec/sqlite-vec.c
+++ b/third_party/sqlite3/ext/sqlite-vec/sqlite-vec.c
@@ -10,6 +10,7 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/types.h>
 
 #ifndef SQLITE_VEC_OMIT_FS
 #include <stdio.h>
