diff --git a/third_party/libwebrtc/modules/desktop_capture/desktop_capturer.h b/third_party/libwebrtc/modules/desktop_capture/desktop_capturer.h
index 2ca2ddc..76e72b8 100644
--- a/third_party/libwebrtc/modules/desktop_capture/desktop_capturer.h
+++ b/third_party/libwebrtc/modules/desktop_capture/desktop_capturer.h
@@ -11,6 +11,9 @@
 #ifndef MODULES_DESKTOP_CAPTURE_DESKTOP_CAPTURER_H_
 #define MODULES_DESKTOP_CAPTURE_DESKTOP_CAPTURER_H_
 
+// pid_t
+#include <unistd.h>
+
 #include <cstddef>
 #include <cstdint>
 #include <memory>
