diff --git a/mesonbuild/mformat.py b/mesonbuild/mformat.py
index 2131ff7b3167..9e1da7292713 100644
--- a/mesonbuild/mformat.py
+++ b/mesonbuild/mformat.py
@@ -242,6 +242,10 @@ def visit_ArgumentNode(self, node: mparser.ArgumentNode) -> None:
         if node.is_multiline:
             self.is_multiline = True
 
+        nargs = len(node)
+        if nargs and nargs == len(node.commas):
+            self.is_multiline = True
+
         if self.is_multiline:
             return
 
diff --git a/test cases/format/1 default/indentation.meson b/test cases/format/1 default/indentation.meson
index fe78847f7667..f18b2307252c 100644
--- a/test cases/format/1 default/indentation.meson	
+++ b/test cases/format/1 default/indentation.meson	
@@ -92,3 +92,11 @@ if meson.project_version().version_compare('>1.2')
         # comment
     endif
 endif
+
+# Test for trailing comma:
+m = [1, 2, 3]
+n = [
+    1,
+    2,
+    3,
+]
